packages/cmrjs/tests/cmr-utils/test-cmr-utils.js

```diff
@@ -729,3 +733,87 @@ test(
   'metadataObjectFromCMRFile throws PreconditionFailed when ETag does not match CMR UMMG JSON metadata file',
   testMetadataObjectFromCMRFile('s3://bucket/fake.cmr.json')
 );
+
+test.serial('publish2CMR passes cmrRevisionId to publishECHO10XML2CMR and receives a revision ID in the log', async (t) => {
+  // const cmrFileObject = { filename: 'test.cmr.xml', granuleId: 'testGranuleId' };
+  const bucket = randomString();
+  const payloadPath = path.join(__dirname, 'data', 'payload.json');
+  const rawPayload = fs.readFileSync(payloadPath, 'utf8');
+  const payload = JSON.parse(rawPayload);
+  const xmlFile = payload.input.granules[0].files[3];
+  const xmlGranuleId = payload.input.granules[0].granuleId;
+  const xmlKey = `${xmlGranuleId}.cmr.xml`;
+  const echoMetadataString = fs.readFileSync(path.join(__dirname, 'data/meta.xml'));
+  const cmrRevisionId = 12;
+  const credentials = await getCmrSettings();
+
+  await createBucket(bucket);
+  await s3PutObject({
+    Bucket: bucket,
+    Key: xmlKey,
+    Body: echoMetadataString,
+  });
+
+  const conceptId = 'id204842';
+  const metadataObject = await metadataObjectFromCMRFile(buildS3Uri(bucket, xmlKey));
+  const updatedXmlFile = { ...xmlFile, granuleId: xmlGranuleId, metadataObject };
+  const logInfoSpy = sinon.spy(log, 'info');
+  const ingestGranuleSpy = sinon.stub(CMR.prototype, 'ingestGranule').returns({ result: { 'concept-id': conceptId, 'revision-id': cmrRevisionId } });
```

Comment:
So I think that this test (and really unit tests) in general should only spy on their direct dependencies. So this test should only verify that it passes the revision ID into `publishECHO10XML2CMR`, not `ingestGranule` which is used by `publishECHO10XML2CMR`. Then it is the job of the unit test for `publishECHO10XML2CMR` to verify that it passes the revision ID into `ingestGranule`. 

I'm sure we're breaking this rule all over the place, but this should be the goal. Otherwise you're unit testing the dependencies of your code, not the unit of code itself.